analyzers/tests/SonarAnalyzer.UnitTest/ControlFlowGraph/ControlFlowGraphTest.cs

```diff
@@ -177,6 +173,52 @@ public class Bar
             VerifyAllInstructions(exit);
         }
 
+        [TestMethod]
+        [TestCategory("CFG")]
+        public void Cfg_ExtremelyNestedExpression_NotSupported_FromExpression()
+        {
+            var method = CompileWithMethodBody(string.Format(TestInput, $"var x = {ExtremelyNestedExpression()};"), "Bar", out var semanticModel);
+            Action a = () => CSharpControlFlowGraph.Create(method.Body, semanticModel);
+
+            a.Should().Throw<NotSupportedException>().WithMessage("Too complex expression");
+            CSharpControlFlowGraph.TryGet(method.Body, semanticModel, out _).Should().BeFalse();
+        }
+
+        [TestMethod]
+        [TestCategory("CFG")]
+        public void Cfg_ExtremelyNestedExpression_NotSupported_FromBodyMethod()
```

Comment:
```suggestion
        public void Cfg_ExtremelyNestedExpression_NotSupported_FromBodyMethodReturn()
```

Reply:
`FromArrowMethod` and `FromBodyMethod` makes a symmetic pair in the list.